command/command.go

```diff
@@ -160,36 +162,50 @@ func (e *ExecutionRequest) Execute(ctx context.Context, execution ExecutionReque
 		resp.Duration = time.Since(started).Seconds()
 	}()
 
-	var timer *time.Timer
-	// Kill process and all of its children when the timeout has expired.
+	timer := time.NewTimer(math.MaxInt64)
+	defer timer.Stop()
 	if execution.Timeout != 0 {
 		SetProcessGroup(cmd)
-		timer = time.AfterFunc(time.Duration(execution.Timeout)*time.Second, func() {
-			timeout()
-			if err := KillProcess(cmd); err != nil {
-				logger.WithError(err).Errorf("Execution timed out - Unable to TERM/KILL the process: #%d", cmd.Process.Pid)
-				escapeZombie(&execution)
-			}
-		})
-		defer timer.Stop()
+		timer.Stop()
+		timer = time.NewTimer(time.Duration(execution.Timeout) * time.Second)
 	}
-
 	if err := cmd.Start(); err != nil {
-		// Something unexpected happended when attepting to
+		// Something unexpected happened when attempting to
 		// fork/exec, return immediately.
 		return resp, err
 	}
 
-	err := cmd.Wait()
-	if timer != nil {
-		timer.Stop()
+	waitCh := make(chan struct{})
+	var err error
+	go func() {
+		err = cmd.Wait()
```

Comment:
It seems this error is unchecked

Reply:
Checked when the process is not killed. I don't believe the error is relevant when the process is killed since we already log and display the kill information.